client/mysqlcheck.c

```diff
@@ -762,7 +762,9 @@ static int handle_request_for_tables(char *tables, size_t length)
   }
   if (mysql_real_query(sock, query, query_length))
   {
-    sprintf(message, "when executing '%s TABLE ... %s'", op, options);
+    const char format_msg[]= "when executing '%s TABLE ... %s'";
+    char message[sizeof(options) + sizeof(format_msg)];
```

Comment:
Strictly speaking `sizeof(options) + sizeof(format_msg) + sizeof(<max_op_len>)`, where `<op>` is one of the following:
```
CHECK
REPAIR
REPAIR NO_WRITE_TO_BINLOG
ANALYZE
ANALYZE NO_WRITE_TO_BINLOG
OPTIMIZE
OPTIMIZE NO_WRITE_TO_BINLOG
```